src/hotspot/share/prims/jvm.cpp

```diff
@@ -2142,14 +2146,27 @@ JVM_ENTRY(jobjectArray, JVM_GetPermittedSubclasses(JNIEnv* env, jclass current))
         if (PENDING_EXCEPTION->is_a(SystemDictionary::VirtualMachineError_klass())) {
           return NULL; // propagate VMEs
         }
+        if (log_is_enabled(Trace, class, sealed)) {
+          stringStream ss;
+          char* permitted_subclass = ik->constants()->klass_name_at(cp_index)->as_C_string();
+          ss.print(" - resolution of permitted subclass %s failed: ", permitted_subclass);
+          java_lang_Throwable::print(PENDING_EXCEPTION, &ss);
+          log_trace(class, sealed)("%s", ss.as_string());
+        }
+
         CLEAR_PENDING_EXCEPTION;
         continue;
       }
       if (k->is_instance_klass()) {
         result->obj_at_put(count++, k->java_mirror());
+        log_trace(class, sealed)(" - [%d] = %s", count, k->external_name());
       }
     }
     if (count < length) {
+      // we had invalid entries so we need to compact the array
+      log_trace(class, sealed)(" - compacting array from length %d to %d",
+                               length, count);
```

Comment:
Not sure this really adds any useful information, but your call.

Reply:
I agree that it is not useful.  I've removed it.
Thanks, Harold